sound/soc/sof/ipc4-topology.c

```diff
@@ -517,10 +517,12 @@ static int sof_ipc4_widget_setup_comp_dai(struct snd_sof_widget *swidget)
 
 	pipe_widget = swidget->spipe->pipe_widget;
 	pipeline = pipe_widget->private;
-	if (pipeline->use_chain_dma && ipc4_copier->dai_type != SOF_DAI_INTEL_HDA) {
+	if (pipeline->use_chain_dma &&
+	    (ipc4_copier->dai_type != SOF_DAI_INTEL_HDA) ||
```

Comment:
should this be &&?

Reply:
yes indeed. Need more coffee and a new brain.